### PR TITLE
Use ffpmeg to convert audio and send to Google Speech API

### PR DESCRIPTION
**Pull request by djbb7**
Audio files are downloaded and converted locally, and then sent to Google.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -19,12 +30,71 @@ app
...
+    })
+    .on('end', function() {
+      //file downloaded, send to Google Speech
+      syncRecognize(output, function(results){
+        //results is the string containing the interpreted audio
+        console.log(results);
+
+        //TODO: hamming distance to find closest match?
+
+        //TODO: respond with closest match
```
Comment (bodyflex) [inline index.js]: are you going to do these in this PR or in a future one?

**Comment by bodyflex** [comment]
thanks for the contribution! why did you change the code to ES5? did you have problems running the ES6 syntax?

File: index.js (hunk 1 above)
Comment (djbb7) [inline index.js]: In a future one

File: restaurants.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
+import distance from "./distance"
```
Comment (bodyflex) [inline restaurants.js]: where should this point to? I don't see any `distance.js` file!

File: restaurants.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+		var index = -1;
+		for (let i = 0; i < this.phonemes.length; i++) {
+			for (let j = 0; j < this.phonemes[i].soundsLike.length; j++) {
+				let lDistance = distance(str, this.phonemes[i].soundsLike[j]);
+				if ( lDistance < bestMatch ){
+					bestMatch = lDistance;
+					index = i;
+				}
+			}
+		}
```
Comment (bodyflex) [inline restaurants.js]: by the way, [lodash](https://lodash.com/docs/4.16.6#minBy) is a great utility library with which this function could be rewritten like this for example:
```js
getClosestMatch(input) {
  return _.minBy(this.phonemes, phoneme => _.min(
    phoneme.soundsLike.map(sound => distance(input, sound))
  ));
}
```
especially when combined with the power of ES6 and functional programming it's really satisfying!

**Pull request closed without merging**